src/Chapter2.hs

```diff
@@ -842,7 +865,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l = if n < 0 then [] else take (length l) (drop n (cycle l))
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. There is a way to optimize this (with `div` 😉 ) :slightly_smiling_face: